src/System.Private.ServiceModel/tests/Scenarios/Extensibility/WebSockets/WebSocketTests.cs

```diff
@@ -134,11 +139,18 @@ public static void WebSocket_Http_Duplex_BinaryStreamed()
 
             client.StopPushingStream();
             // Waiting on ReceiveStreamCompleted from the ClientReceiver.
-            clientReceiver.ReceiveStreamCompleted.WaitOne();
+            Assert.True(clientReceiver.ReceiveStreamCompleted.WaitOne(ScenarioTestHelpers.TestTimeout),
+                "Test case timeout was reached while waiting for the stream response from the Service to be completed.");
             clientReceiver.ReceiveStreamCompleted.Reset();
 
+            // Getting results from server via callback.
+            client.GetLog();
+            Assert.True(clientReceiver.LogReceived.WaitOne(ScenarioTestHelpers.TestTimeout),
```

Comment:
It doesn't reduce the time for each WaitOne call, if everything is working as expected the call should complete well before the timeout, it was just recommended that we wrap the WaitOne call this way 'just in case' the call ever failed for some reason so that the test wouldn't hang here for a long time until some other timeout gets hit.